src/Nethermind/Nethermind.Consensus/Processing/BlockExtensions.cs

```diff
@@ -19,7 +19,7 @@ internal static class BlockExtensions
         public static Block CreateCopy(this Block block, BlockHeader header) =>
             block is BlockToProduce blockToProduce
                 ? new BlockToProduce(header, blockToProduce.Transactions, blockToProduce.Uncles, blockToProduce.Withdrawals)
```

Comment:
@Demuirgos @rjnrohit check this line carefully:
`                ? new BlockToProduce(header, blockToProduce.Transactions, blockToProduce.Uncles, blockToProduce.Withdrawals)	`

analyze all usages of BlockToProduce constructor